### ISSUE TITLE
Does Skillet offer leveling optimization?

### ISSUE THREAD

**Issue by brandonsturgeon** [b-morgan/Skillet#49]
I've been dreaming of an addon that does everything Skillet does, but with only one thing (apparently?) missing; my dream profession addon would also estimate how many of a given recipe I should make before it's more efficient to move on to another recipe.

Is this something that Skillet offers already?

If not, have you spent any time working on such a feature? I'd be happy to take a stab at adding it, or something like it.

Thank you for maintaining such a useful addon!

**Comment by b-morgan** [comment]
Skillet does not have such an estimator and I believe it is beyond the scope of the addon. Blizzard does not provide an API to get accurate skill up data so Skillet has to rely on other sources for that data. Skillet does have its own table (in SkillLevelData.lua) because someone other than myself collected the data by scrapping Wowhead with ScrapeWowhead.js.

Once a recipe turns Yellow or Green, the RNG comes into play so there is only a probability of a skill up, not a certainty. The formula for this probability is implemented in Skillet and can be seen if you mouse over the skill up bar (and it assumes accurate skill up data from the table).

What Blizzzard does provide is the color of the recipe and how many skill points per craft and both of those are displayed by Skillet.

If you wish to contribute, you could use the Javascript code and scrape Wowhead for any updated data. I believe this data would be necessary for adding any additional features to Skillet.

**Comment by brandonsturgeon** [comment]
Understood - that makes sense.

In my mind, I think a best-guess estimator would be sufficient. It doesn't have to be perfect, as even having a rough estimate would be enormously useful.

For example, yesterday I was leveling engineering. I had a single orange recipe and I wanted to make that recipe until it was no longer economical to do so, but I have no idea how much that actually is.
I was running back and forth between the AH and the anvil until it finally turned green.

I know, I know, that's kind of the WoW experience, but I think it could be made less tedious with an addon's instructions.

In conclusion, however, I agree - this feature might be out of scope for an addon like Skillet. If some third party site or tool was available ( [professions.gg](https://wotlk.professions.gg/) but for retail, for example ) that instructed users how many of each recipe to make, Skillet would be a stellar companion tool.

Thanks for your prompt, informative response - it's much appreciated!

**Comment by b-morgan** [comment]
I've found that the leveling guides on Wowhead are pretty good at telling you the optimum way to level.

For Orange recipes, either Wowhead or Skillet itself can tell you at what level it goes Yellow. That will tell you how many you can make to get to that point. Then I go to the trainer to see if I can learn something else that's Orange before I keep making Yellow recipes and I avoid making Green recipes unless I'm desperate, lol.

**Comment by brandonsturgeon** [comment]
Haha, completely understandable.

If you could point me to the pages you're referencing, I could whip up a convenient scraper. I have a lot of experience with web scraping. I could even set it up such that it scrapes and updates the data files automatically - perhaps when you merge a PR or tag a new release.

**Comment by b-morgan** [comment]
The file `...\Interface\Addons\Skillet-Classic\ScrapeWowhead.js` contains some instructions at the beginning and at the end including an example page (the other pages should be obvious). Adjust the page references to Wrath instead of TBC.

Feel free to either modify the existing code or start fresh. Please add instructions for use as comments. You can upload the code used to scrape and the resulting data file here. I'll incorporate both into the next build.
